### PR TITLE
Remove unwrap

### PR DESCRIPTION
**Pull request by bitfriend**
Will replace `unwrap()` with `expect()` or `context()`.

### CODE REVIEW

File: native/acter/src/api/stream.rs [hunk 1]
```diff
@@ -411,7 +412,7 @@ impl TimelineStream {
 
     pub async fn retry_send(&self, txn_id: String) -> Result<bool> {
         let timeline = self.timeline.clone();
-        let transaction_id = OwnedTransactionId::from(txn_id);
```
Comment (gnunicorn) [inline native/acter/src/api/stream.rs:414]: this one I don't understand. Why replace a never failing `from` with a sometimes failing `try_from(...)?`?

File: native/acter/src/api/stream.rs [hunk 2]
```diff
@@ -430,15 +431,15 @@ impl TimelineStream {
...
-                timeline.retry_send(&transaction_id).await;
+                timeline.retry_send(&txn_id).await?;
                 Ok(true)
             })
             .await?
     }
 
     pub async fn cancel_send(&self, txn_id: String) -> Result<bool> {
         let timeline = self.timeline.clone();
-        let transaction_id = OwnedTransactionId::from(txn_id);
```
Comment (gnunicorn) [inline native/acter/src/api/stream.rs:441]: same as above...

File: native/acter/src/api/stream.rs [hunk 3]
```diff
@@ -457,7 +458,7 @@ impl TimelineStream {
                     bail!("No permission to send message in this room");
                 }
 
-                timeline.cancel_send(&transaction_id).await;
```
Comment (gnunicorn) [inline native/acter/src/api/stream.rs:460]: why does this not get a `?`?

File: native/acter/src/api/verification.rs [hunk 4]
```diff
@@ -1066,7 +1070,7 @@ impl VerificationController {
                                 msg.set_content("short_authentication_string".to_string(), short_authentication_string.join(","));
                             }
                             StartMethod::ReciprocateV1(content) => {
-                                let secret = serde_json::to_string(&content.secret).unwrap();
+                                let secret = serde_json::to_string(&content.secret).expect("couldn't convert secret to string");
```
Comment (gnunicorn) [inline native/acter/src/api/verification.rs:1073]: considering that this is in a callback, we probably don't want to panic here but instead do log if this doesn't work...
```suggestion
                            
let secret = match serde_json::to_string(&content.secret) {
    Ok(e) => e;
    Err(e) => {
       error!("Parsing RecipV1 secret failed", error);
       return;
    }
};
```

File: native/acter/src/api/verification.rs [hunk 5]
```diff
@@ -1151,7 +1155,7 @@ impl VerificationController {
                             evt.sender,
                         );
                         msg.set_content("keys".to_string(), evt.content.keys.to_string());
-                        let mac = serde_json::to_string(&evt.content.mac).unwrap();
+                        let mac = serde_json::to_string(&evt.content.mac).expect("couldn't convert mac to string");
```
Comment (gnunicorn) [inline native/acter/src/api/verification.rs:1158]: same logging as above instead of panic...

**Review by gnunicorn — APPROVED** [review]
Nice. I like replacing those `else bail` with `.context()` makes for smaller less expanded code blocks. Just one remark on an `expect` in a callback that we probally don't want to crash on if that failed.

File: native/acter/src/api/stream.rs (hunk 1 above)
Comment (bitfriend) [reply native/acter/src/api/stream.rs:414]: I just wanted to capture incorrect usage in development stage.

File: native/acter/src/api/stream.rs (hunk 3 above)
Comment (bitfriend) [reply native/acter/src/api/stream.rs:460]: `Timeline::cancel_send()` doesn't return `Result<>`.

File: native/acter/src/api/verification.rs (hunk 4 above)
Comment (bitfriend) [reply native/acter/src/api/verification.rs]: Fixed by https://github.com/acterglobal/a3/pull/1245/commits/a183faf7571b2a83fa1b0d3e33d6a15d2fb4f03b

File: native/acter/src/api/verification.rs (hunk 5 above)
Comment (bitfriend) [reply native/acter/src/api/verification.rs]: Fixed by https://github.com/acterglobal/a3/pull/1245/commits/a183faf7571b2a83fa1b0d3e33d6a15d2fb4f03b

File: native/acter/src/api/stream.rs (hunk 1 above)
Comment (bitfriend) [reply native/acter/src/api/stream.rs:414]: And I don't know why both `from()` and `try_from()` exist in `OwnedTransactionId`.
If failure is possible like `OwnedEventId` or `OwnedUserId`, `from()` should not exist.

**Pull request merged**